store/kv.go

```diff
@@ -41,15 +41,17 @@ func PrefixEntries(ctx context.Context, store ds.Datastore, prefix string) (dsq.
 }
 
 // GenerateKey ...
-func GenerateKey(fields []interface{}) string {
-	var b bytes.Buffer
+func GenerateKey(fields []string) (string, error) {
+	var b strings.Builder
 	b.WriteString("/")
 	for _, f := range fields {
-		b.Write([]byte(fmt.Sprintf("%v", f) + "/"))
+		_, err := b.WriteString(f + "/")
+		if err != nil {
+			return "", err
+		}
 	}
-	return path.Clean(b.String())
+	return path.Clean(b.String()), nil
```

Comment:
Since `fields` are strings now, you can as well just call `strings.Join` for simplicity.

```go
key := "/" + strings.Join(fields, "/")
return path.Clean(key)
 ```
or even:
```go
func GenerateKey(fields ...string) string {
	return path.Join(fields...)
}
```
(leading `/` probably can be sacrificed without issues)